devito/operator.py

```diff
@@ -190,6 +190,11 @@ def __init__(self, expressions, **kwargs):
         iet = self._finalize(iet, parameters)
 
         super(Operator, self).__init__(self.name, iet, 'int', parameters, ())
+        config_state = {}
```

Comment:
I think we should make this such that it can be specialised by an Operator subclass. That is, we should put this in a method and replace ``self._state = {}`` a few lines above with, for example, ``self._state = self._initialize_state(...)``

then you have
```
def _initialize_state(self, **kwargs):
    # equivalent to what you wrote, but more pythonic and homogeneous with the rest of devito
    return {i: kwargs.get(i, v) for k, v in configuration.items()}
```

Reply:
Submitted a new commit with this kind of change, except the state dictionary stores configuration items as a nested dictionary, so that the keys in the configuration dictionary at construction and the pre-existing keys used in the state dictionary (i.e. 'autotuning') don't interact.